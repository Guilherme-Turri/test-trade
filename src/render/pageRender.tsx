import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};