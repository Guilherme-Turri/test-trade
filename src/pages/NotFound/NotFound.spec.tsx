import { pageRender } from "../../render/pageRender";
import { NotFound } from ".";
import { screen } from '@testing-library/react'

describe('<NotFound>', () => {
  it('should render on screen properly', () => {
    const { container } = pageRender(<NotFound />)
    const notFound = screen.getByTestId('NotFoundLocation')
    expect(notFound).toBeInTheDocument();
    expect(container).toMatchSnapshot();

  })
  it('should navigate properly', () => {
    pageRender(<NotFound />)
    const link = screen.getByRole('link')
    expect(link).toHaveAttribute('href', '/')
  })
})