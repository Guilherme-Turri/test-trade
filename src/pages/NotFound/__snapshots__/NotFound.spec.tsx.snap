// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound> should render on screen properly 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c3 {
  color: #00c555;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
}

.c0 a {
  margin-top: 0.938rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ff5555;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  z-index: 99999999;
}

.c2 {
  height: 15rem;
  width: 15rem;
}

.c6 {
  color: #F3F4F5;
  font-weight: bold;
  padding: 0.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  background-image: url(wave.svg);
  background-repeat: no-repeat;
  background-size: cover;
  width: 100%;
  height: 20rem;
  z-index: -10000;
}

.c5 {
  text-transform: uppercase;
  color: #00c555;
  font-size: 60px;
  font-family: 'Bruno Ace SC',cursive;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          data-testid="NotFoundLocation"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M91.967 7.961c0-.016.005-.031.005-.047a2.733 2.733 0 0 0-2.73-2.733H39.559v.011L8.031 36.721h-.003v55.365h.003v.001a2.735 2.735 0 0 0 2.734 2.731h78.479a2.73 2.73 0 0 0 2.663-2.15h.06v-.536c0-.015.004-.029.004-.044s-.004-.029-.004-.044V7.961zm-24.328 7.177H82.01v24.597L63.897 21.621l3.742-6.483zM39.57 39.453v-.001a2.732 2.732 0 0 0 2.722-2.73V15.138H61.88l-27.17 47.06-16.725-16.725v-6.02H39.57zM17.985 84.862V52.527L32.128 66.67 21.626 84.862h-3.641zm9.4 0 33.93-58.769 20.696 20.696v38.073H27.385z"
            />
            <path
              d="M62.03 45.576c-6.645 0-12.026 5.387-12.026 12.027 0 2.659.873 5.109 2.334 7.1l7.759 13.439c.047.094.097.186.157.271l.016.027.004-.002a2.16 2.16 0 0 0 3.405.132l.02.011.075-.129a2.25 2.25 0 0 0 .287-.497l7.608-13.178a11.962 11.962 0 0 0 2.39-7.175c-.003-6.639-5.384-12.026-12.029-12.026zM61.911 63.7a5.924 5.924 0 0 1-5.926-5.925 5.926 5.926 0 1 1 5.926 5.925z"
            />
          </svg>
        </div>
      </div>
      <h1
        class="c5"
      >
        404
      </h1>
      <a
        href="/"
      >
        Pagina nao encontrada! Voltar
      </a>
    </div>
    <footer
      class="c6"
    >
      Guilherme Turri - Trade Technology
    </footer>
  </section>
</div>
`;
